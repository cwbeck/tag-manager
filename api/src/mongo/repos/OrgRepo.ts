import Repo from '../abstractions/Repo';
import { injectable } from 'inversify';
import Org from '../models/Org';
import { IndexSpecification } from 'mongodb';

@injectable()
export default class OrgRepo extends Repo<Org> {
    protected readonly auditEnabled = true;

    protected readonly indexes: IndexSpecification[] = [
        {
            background: false,
            key: {
                _name: 1,
            },
            unique: true,
        },
    ];
}
