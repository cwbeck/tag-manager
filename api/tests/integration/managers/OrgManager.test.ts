import ManagerTestUtils from '../../utils/ManagerTestUtils';
import TestUtils from '../../utils/TestUtils';
import OrgRepo from '../../../src/mongo/repos/OrgRepo';
import Mock = jest.Mock;

describe('OrgManager', () => {
    beforeEach(ManagerTestUtils.beforeEachPrepareContainer);
    afterEach(TestUtils.afterEachRestoreContainerMocksAndDate);

    it('fetches single org', async () => {
        ManagerTestUtils.mockOrgUserWithViewAccess();

        const mockSutModel = TestUtils.buildMockOrg();
        const mockSutRepo = {} as OrgRepo;
        mockSutRepo.findByIdThrows = jest.fn().mockResolvedValue(mockSutModel);
        (ManagerTestUtils.mockRepoFromModelFactory as Mock).mockReturnValue(mockSutRepo);

        const server = TestUtils.createApolloServerForIntegration();

        const Q = `
            query {
                getOrg(id: "5edf4c48e9615d9a1c66bc2a") {
                    id
                    name
                }
            }
        `;

        // run query against the server and snapshot the output
        const res = await server.executeOperation({ query: Q });
        expect(res).toMatchSnapshot();
    });
});
