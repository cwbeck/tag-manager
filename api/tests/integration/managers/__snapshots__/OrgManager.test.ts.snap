// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrgManager fetches single org 1`] = `
Object {
  "data": Object {
    "getOrg": Object {
      "id": "5edf4c48e9615d9a1c66bbbb",
      "name": "testOrg",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
